File: buffer/cpu/src/jvmMain/java/com/wsr/cpu/elementwise/compare/where/JWhere.java
package com.wsr.cpu.elementwise.compare.where;

import java.nio.ByteBuffer;

public class JWhere {
    public native void whereD0ToD0(ByteBuffer condition, float x, float y, ByteBuffer result);
    public native void whereD0ToD1(ByteBuffer condition, float x, ByteBuffer y, ByteBuffer result);
    public native void whereD1ToD0(ByteBuffer condition, ByteBuffer x, float y, ByteBuffer result);
    public native void whereD1ToD1(ByteBuffer condition, ByteBuffer x, ByteBuffer y, ByteBuffer result);
}

File: buffer/cpu/src/jvmAndroidMain/kotlin/com/wsr/knist/cpu/elementwise/compare/where/JWhere.kt
package com.wsr.knist.cpu.elementwise.compare.where

object JWhere {
    external fun whereD0ToD0(condition: Long, x: Float, y: Float, result: Long)
    external fun whereD0ToD1(condition: Long, x: Float, y: Long, result: Long)
    external fun whereD1ToD0(condition: Long, x: Long, y: Float, result: Long)
    external fun whereD1ToD1(condition: Long, x: Long, y: Long, result: Long)
}
